### PR TITLE
Netbsd pep

### PR DESCRIPTION
**Pull request by FredericHamel**

### CODE REVIEW

File: lib/os_files.c [hunk 1]
```diff
@@ -1568,21 +1568,23 @@ ___SCMOBJ ___os_executable_path ___PVOID
 #if defined (USE_sysctl) && defined (CTL_KERN) && defined (KERN_PROC) && defined (KERN_PROC_PATHNAME)
 
   {
-    int mib[4] =
+    // Each row has the format: nb_args, arg0, ..., argn
```
Comment (feeley) [inline lib/os_files.c]: should be: nb_args, arg1, ..., argn

File: lib/os_files.c [hunk 2]
```diff
@@ -1568,21 +1568,23 @@ ___SCMOBJ ___os_executable_path ___PVOID
...
 #if defined(KERN_PROC_ARGS)
-      { CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME };
-#else
-      { CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1 };
+      4, CTL_KERN, KERN_PROC_ARGS, -1, KERN_PROC_PATHNAME,
 #endif
+      4, CTL_KERN, KERN_PROC, KERN_PROC_PATHNAME, -1,
+      0
+    };
+    int *mibs_iter = (int*)mibs;
```
Comment (feeley) [inline lib/os_files.c]: use mibs_probe instead of mibs_iter, and remover (int*) cast which is redundant.

**Comment by feeley** [comment]
Something went wrong... you reverted to the original os_files.c file.

**Comment by FredericHamel** [comment]
I don't what went wrong. The commit changes are impossible.

**Comment by FredericHamel** [comment]
Don't merge!!! There is something wrong with this.

**Comment by feeley** [comment]
Too late! Please fix and do another pull request.

**Comment by FredericHamel** [comment]
The history was broken.

**Pull request merged**